bitbot/markov.go

```diff
@@ -46,3 +74,31 @@ func generateBabble(chain *gomarkov.Chain) string {
 	}
 	return strings.Join(tokens[1:len(tokens)-1], " ")
 }
+
+func markovInit(chain *gomarkov.Chain) bool {
+	var sources = [3]string{"https://gist.githubusercontent.com/bbriggs/60e907f3571a1ca7c41cd99f78052d78/raw/fe6d0bd96ee97c9b5df2794ae683d24a404b4433/bible.txt",
+		"https://gist.githubusercontent.com/bbriggs/f63340a3ed1a1439b6f3f8d619eacac1/raw/1f363d500226c55bab735fe59074f06721348546/world_factbook.txt",
+		"https://gist.githubusercontent.com/parsec/2f4d4edf55336c0a2994cfcf951a8ea7/raw/4b66c99f1879b927ebc2b2ffb8fdd39dc9a4f7d2/SnwCrsh"}
+
+	i := 0
+	for i <= len(sources) {
+		resp, err := http.Get(sources[i])
+		if err != nil {
+			fmt.Println(err)
+			return false
+		}
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		bodyString := string(body)
+		markovAdd(bodyString, chain)
+	}
+	return true
+}
+
+// wrapper for b.mChain.Add that includes file lock/unlock
+func markovAdd(text string, chain *gomarkov.Chain) {
```

Comment:
maybe we should make the mutex an argument as well so that this function is properly isolated